runtime/connectors/blob/blobdownloader.go

```diff
@@ -239,8 +245,24 @@ func (it *blobIterator) plan() ([]*objectWithPlan, error) {
 		return nil, err
 	}
 
-	listOpts := listOptions(it.opts.GlobPattern)
+	listOpts, singleFile := listOptions(it.opts.GlobPattern)
```

Comment:
This would be more intuitive with a regular go-like `val, ok :=` pattern, like `listOpts, ok := `

Reply:
Makes sense. I was actually looking for a better way to do this and added this at last resort.